plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFENotCanonical.sol

```diff
@@ -85,6 +91,35 @@ library PaymentChallengeIFENotCanonical {
         emit InFlightExitChallenged(msg.sender, keccak256(args.inFlightTx), competitorPosition);
     }
 
+    function respond(
+        Controller memory self,
+        PaymentExitDataModel.InFlightExitMap storage inFlightExitMap,
+        bytes memory inFlightTx,
+        uint256 inFlightTxPos,
+        bytes memory inFlightTxInclusionProof
+    )
+        public
+    {
+        uint192 exitId = ExitId.getInFlightExitId(inFlightTx);
```

Comment:
> In case when we are using calls to libraries from multiple places, there can happen a situation when part of the code is deployed with one library linked and another part with a different (let's say malicious) version.

I am not fully understanding this comment. The `ExitId` library is currently designed to not need to be linked (as functions are `internal`), so what is the concern here?

Reply:
Oh, sorry, I was not aware of that. I had in my mind the same model that worked for RootChain

Reply:
As off-site discussion emerged, we decided to create an issue (#253) to rethink remedy for `code duplication` (not related to the :point_up:  original comment)